crates/biome_css_analyze/src/lint/nursery/no_font_family_duplicate_names.rs

```diff
@@ -0,0 +1,200 @@
+        version: "next",
+        name: "noFontFamilyDuplicateNames",
+        recommended: true,
+        source: RuleSource::Stylelint("font-family-no-duplicate-names"),
+    }
+}
+
+pub struct RuleState {
+    value: String,
+    span: TextRange,
+}
+
+impl Rule for NoFontFamilyDuplicateNames {
+    type Query = Ast<CssGenericProperty>;
+    type State = RuleState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Option<Self::State> {
+        let node = ctx.query();
+        let property_name = node.name().ok()?.text().to_lowercase();
+
+        let is_font_family = property_name == "font-family";
+        let is_font = property_name == "font";
+
+        if !is_font_family && !is_font {
+            return None;
+        }
+
+        let mut seen: HashSet<String> = HashSet::new();
+        let value_list = node.value();
+        let font_families = if is_font {
+            find_font_family(value_list)
+        } else {
+            value_list
+                .into_iter()
+                .filter_map(|v| match v {
+                    AnyCssGenericComponentValue::AnyCssValue(value) => Some(value),
+                    _ => None,
+                })
+                .collect()
+        };
+
+        for css_value in font_families {
+            match css_value {
+                // A generic family name like `serif` or `sans-serif`
+                AnyCssValue::CssIdentifier(val) => {
+                    let font_name = val.text();
+                    if seen.contains(&font_name) {
+                        return Some(RuleState {
+                            value: font_name.clone(),
+                            span: val.range(),
+                        });
+                    }
+                    seen.insert(font_name);
+                }
+                // A font family name. e.g "Lucida Grande", "Arial".
+                AnyCssValue::CssString(val) => {
+                    let normalized_font_name: String = val
+                        .text()
+                        .chars()
+                        .filter(|&c| c != '\'' && c != '\"' && !c.is_whitespace())
+                        .collect();
+
+                    if seen.contains(&normalized_font_name) {
+                        return Some(RuleState {
+                            value: normalized_font_name,
+                            span: val.range(),
+                        });
+                    }
+                    seen.insert(normalized_font_name);
+                }
+                _ => continue,
+            }
+        }
+        None
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let span = state.span;
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            span,
+            markup! {
+                "Unexpected duplicate font name: "<Emphasis>{ state.value }</Emphasis>
+            },
+        ))
```

Comment:
Let's follow the [rule pillars](https://biomejs.dev/linter/#rule-pillars):
- show the error
- explain why it could be an error
- suggest a solution

The solution usually is the code action, if we don't have a code action, we should suggest something.